cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -657,34 +986,85 @@ class AsofJoinNode : public ExecNode {
     return std::make_shared<arrow::Schema>(fields);
   }
 
+  static inline Result<col_index_t> FindColIndex(const Schema& schema,
+                                                 const FieldRef& field_ref,
+                                                 const util::string_view& key_kind) {
+    auto match_res = field_ref.FindOne(schema);
+    if (!match_res.ok()) {
+      return Status::Invalid("Bad join key on table : ", match_res.status().message());
+    }
+    auto match = match_res.ValueOrDie();
```

Comment:
Don't think we can use `ValueOrDie()` here - can we pop the error?

Reply:
The preceding `if`-block guarantees `match_res` has a value, so `match_res.ValueOrDie()` will never raise an error.